source/Engine.cpp

```diff
@@ -2020,15 +2026,30 @@ void Engine::DoCollisions(Projectile &projectile)
 	}
 	else if(projectile.MissileStrength())
 	{
-		// If the projectile did not hit anything, give the anti-missile systems
-		// a chance to shoot it down.
+		// If the projectile did not hit anything, check for anti-missile
+		// systems in range and record their distances.
 		for(Ship *ship : hasAntiMissile)
 			if(ship == projectile.Target() || gov->IsEnemy(ship->GetGovernment()))
-				if(ship->FireAntiMissile(projectile, visuals))
-				{
-					projectile.Kill();
-					break;
-				}
+			{
+				const double distanceSquared = ship->IsInAntiMissileRangeOf(projectile);
+				if (distanceSquared >= 0)
+					antiMissileTargets[ship].emplace(distanceSquared, &projectile);
```

Comment:
Did you mean to change `distanceSquared` to `distance`? The former is more technically accurate.

Reply:
Yes, dropping the const qualifier and changing the name of the very-short-lived local variable were both intentional. That it is a squared distance rather than a "regular' distance is an implementation detail we have chosen to make for the sake of speed, and if we remove the superfluous distance check from `Ship::FireAntiMissile`, we don't even really need to care, as we don't have to pass it to the method at all.

And to be even more truthful to it's purpose, it should probably be called a "rank"--an astute mathematician would remove the `if` statement here, since a squared distance _is always_ greater than or equal to 0.